examples/transforms/global_vs_local_translation.rs

```diff
@@ -0,0 +1,116 @@
+use bevy::prelude::*;
+
+struct SomeEntity {
+    spawn: Vec3,
+    stopped: bool,
+}
+struct GlobalStop;
+struct LocalStop;
+
+const MAX_DISTANCE: f32 = 5.0;
+
+impl SomeEntity {
+    fn new(spawn: Vec3) -> Self {
+        SomeEntity {
+            spawn,
+            stopped: false,
+        }
+    }
+}
+
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup.system())
+        .add_system(move_cubes.system())
+        .add_system(stop_too_far_local_distance.system())
+        .add_system(stop_too_far_global_distance.system())
+        .run();
+}
+
+fn stop_too_far_global_distance(
+    mut cubes: Query<(&GlobalTransform, &mut SomeEntity, &GlobalStop)>,
+) {
+    for (transform, mut cube, _) in cubes.iter_mut() {
+        if (transform.translation - cube.spawn).length() >= MAX_DISTANCE {
+            cube.stopped = true;
+        }
+    }
+}
+
+fn stop_too_far_local_distance(mut cubes: Query<(&Transform, &mut SomeEntity, &LocalStop)>) {
+    // when checking the distance between a point with local transform as reference
+    // we'll get a different outcome due to the fact that this point is in local coordinates
+    // thus stopping the green cube only after reaching a distance of MAX_DISTANCE
+    // to its spawn point above the now moved yellow parent block
+    for (transform, mut cube, _) in cubes.iter_mut() {
+        if (transform.translation - cube.spawn).length() >= MAX_DISTANCE {
+            cube.stopped = true;
+        }
+    }
+}
+
+fn move_cubes(mut cubes: Query<(&mut Transform, &SomeEntity)>, timer: Res<Time>) {
+    for (mut transform, _) in cubes.iter_mut().filter(|(_, cube)| !cube.stopped) {
+        // using the local transform we can move an entity along it's local z axis
+        let dir = transform.local_x();
+        transform.translation += dir * timer.delta_seconds();
+    }
+}
+
+fn setup(
```

Comment:
IMO you should try to mimic the order of your AppBuilder and the functions in your code, so you can easily find them.